Port wiki pages to documentation website

H-Shay opened the pull request:
Fixes #11299. There was one file I did not port, it seemed out-of-date and perhaps not up to "official documentation" standards, but I am happy to change if others disagree.

dklimpel commented on docs/usage/administration/Admin_FAQ.md:
````diff
@@ -0,0 +1,96 @@
+GROUP BY s.canonical_alias, g.room_id
+ORDER BY num_rows desc 
+LIMIT 10;
+```
````
~~~suggestion
```

You can also use the [List Room API](https://matrix-org.github.io/synapse/latest/admin_api/rooms.html#list-room-api)
and `order_by` `state_events`.
~~~

dklimpel commented on docs/usage/administration/Admin_FAQ.md:
```diff
@@ -0,0 +1,96 @@
+---
+Using the following curl command:
+
+    curl -H 'Authorization: Bearer <access-token>' -X DELETE https://matrix.org/_matrix/client/r0/directory/room/<room-alias>
```
~~~suggestion
```
curl -H 'Authorization: Bearer <access-token>' -X DELETE https://matrix.org/_matrix/client/r0/directory/room/<room-alias>
```
~~~

dklimpel commented on docs/usage/administration/Admin_FAQ.md:
```diff
@@ -0,0 +1,96 @@
+
+    curl -H 'Authorization: Bearer <access-token>' -X DELETE https://matrix.org/_matrix/client/r0/directory/room/<room-alias>
+
+\<access-token\> - can be obtained in riot by looking in the riot settings, down the bottom is:
```
```suggestion
`<access-token>` - can be obtained in riot by looking in the riot settings, down the bottom is:
```

dklimpel commented on docs/usage/administration/Admin_FAQ.md:
```diff
@@ -0,0 +1,96 @@
+\<access-token\> - can be obtained in riot by looking in the riot settings, down the bottom is:
+Access Token:\<click to reveal\> 
+
+\<room-alias\> - the room alias, eg. #my_room:matrix.org this possibly needs to be URL encoded also, for example  %23my_room%3Amatrix.org
```
```suggestion
`<room-alias>` - the room alias, eg. #my_room:matrix.org this possibly needs to be URL encoded also, for example  %23my_room%3Amatrix.org
```

dklimpel commented on docs/usage/administration/Synapse-database-maintenance-tools.md:
```diff
@@ -0,0 +1,36 @@
+
+# List of useful tools and scripts for maintenance Synapse database:
+
+## [Purge Remote Media API](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/purge_remote_media.rst)
```
```suggestion
## [Purge Remote Media API](https://matrix-org.github.io/synapse/latest/admin_api/media_admin_api.html#purge-remote-media-api)
```

dklimpel commented on docs/usage/administration/Synapse-database-maintenance-tools.md:
```diff
@@ -0,0 +1,36 @@
+
+#### Purge local media 
+
+There is no purge API for local media because you may be the only one with a copy. If you are sure you want to delete local media you could use something like the following to delete media that hasn't been accessed in a while.
```
This is not up to date.
There is an API: https://matrix-org.github.io/synapse/latest/admin_api/media_admin_api.html#delete-local-media

dklimpel commented on docs/usage/administration/Synapse-database-maintenance-tools.md:
```diff
@@ -0,0 +1,36 @@
+
+**Warning, check noatime flag**: You also have to double-check that the filesystem where synapse's media store don't have [noatime flag](https://unix.stackexchange.com/questions/219015/how-to-disable-access-time-settings-in-debian-linux/219017#219017). Check it with `mount`, `noatime` is something that is usually enabled by default to reduce read-write operations in the filesystem for a feature i ...
+
+## [Purge History API](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/purge_history_api.rst)
```
```suggestion
## [Purge History API](https://matrix-org.github.io/synapse/latest/admin_api/purge_history_api.html)
```

dklimpel commented on docs/usage/administration/Understanding-Synapse-Performance-Issues-Through-Grafana-Graphs.md:
```diff
@@ -0,0 +1,72 @@
+## Understanding Synapse through Grafana graphs
+
+It is possible monitor much of the internal state of Synapse using [Prometheus](https://prometheus.io) metrics and [Grafana](https://grafana.com/). A guide for configuring Synapse to provide metrics is available [here](https://github.com/matrix-org/synapse/blob/master/docs/metrics-howto.md) and information on setting up Grafana is [here](https://github.com/matrix-org/synapse/tree/master/contrib/g ...
```
```suggestion
It is possible monitor much of the internal state of Synapse using
[Prometheus](https://prometheus.io) metrics and [Grafana](https://grafana.com/).
A guide for configuring Synapse to provide metrics is available
[here](https://matrix-org.github.io/synapse/latest/metrics-howto.html)
and information on setting up Grafana is
[here](https://github.com/matrix-org/synapse/tree/master/contrib/grafana).
In this setup, Prometheus will periodically scrape the information Synapse provides and
store a record of it over time. Grafana is then used as an interface to query and
present this information through a series of pretty graphs.
```

dklimpel commented on docs/usage/administration/Understanding-Synapse-Performance-Issues-Through-Grafana-Graphs.md:
```diff
@@ -0,0 +1,72 @@
+
+## Final Thoughts
+
+If you're still having performance problems with your Synapse instance and you've tried everything you can, it may just be a lack of system resources. Consider adding more CPU and RAM, and make use of [worker mode](https://github.com/matrix-org/synapse/blob/master/docs/workers.md) to make use of multiple CPU cores / multiple machines for your homeserver.
```
```suggestion
If you're still having performance problems with your Synapse instance and you've tried
everything you can, it may just be a lack of system resources. Consider adding more CPU
and RAM, and make use of [worker mode](https://matrix-org.github.io/synapse/latest/workers.html)
to make use of multiple CPU cores / multiple machines for your homeserver.
```

anoadragon453 commented:
@dklimpel links leading to other parts of the documentation should be done through relative links to `.md` files. `mdbook` will then generate the right URLs when it builds the documentation.

Additionally, let's not try to review the content too much in this PR. Let's get it in in it's current state, and then iterate smaller fixes to the content from there.

H-Shay commented:
@anoadragon453- is there anything you want me to pull from the review and add, or is this approved to merge in it's current state? Happy to make any changes necessary, just let me know what's best to proceed!

dklimpel commented:
> Additionally, let's not try to review the content too much in this PR. Let's get it in in it's current state, and then iterate smaller fixes to the content from there.

I agree with you. But IMO if the documentation is moved, at least the links should be adjusted.

squahtx commented:
A nit: could we lowercase the file names and convert them to `snake_case`?

anoadragon453 commented:
> @anoadragon453 -is there anything you want me to pull from the review and add, or is this approved to merge in it's current state? Happy to make any changes necessary, just let me know what's best to proceed!

Some things, yes. Sorry for the wait! I'm currently in the process of reviewing, and will request changes/approve then!

anoadragon453 commented on docs/other/Running-synapse-on-Single-board-computers.md:
```diff
@@ -0,0 +1,74 @@
+
+![Screenshot_2020-10-02_17-15-06](https://user-images.githubusercontent.com/71895/94945781-18771500-04d3-11eb-8419-83c2da73a341.png)
+
+This is #1211 and will also hopefully be mitigated by peeking matrix-org/matrix-doc#2753 so at least you don't need to wait for a join to complete before finding out if it's the kind of room you want. Note that you should first disable presence, otherwise it'll just make the situation worse #3120. There is a lot of database interaction too, so make sure you've [migrated your data](https://github. ...
```
```suggestion
This is #1211 and will also hopefully be mitigated by peeking matrix-org/matrix-doc#2753 so at least you don't need to wait for a join to complete before finding out if it's the kind of room you want. Note that you should first disable presence, otherwise it'll just make the situation worse #3120. There is a lot of database interaction too, so make sure you've [migrated your data](../postgres.md) from the default sqlite to postgresql. Personally, I recommend patience - once the initial join is complete there's rarely any issues with actually interacting with the room, but if you like you can just block "large" rooms entirely.
```

anoadragon453 commented on docs/usage/administration/Admin_FAQ.md:
```diff
@@ -0,0 +1,96 @@
+
+What do all those fields in the 'Processed' line mean?
+---
+See https://matrix-org.github.io/synapse/latest/usage/administration/request_log.html.
```
```suggestion
See [Request log format](request_log.md).
```

anoadragon453 reviewed (changes requested):
Thank you for doing this, and apologies again for the delay on review!

Some thoughts about visibility for contributors. Otherwise, I think the only changes to the content that are needed are ensuring all links to other bits of the documentation are relative links so that they render correctly on the site. That, and linking all "#xxxx" issues to the repo. I would leave behind any other fixes to the documentation, such as inaccuracies, and take those already suggested and include them in a subsequent PR. Just to limit the scope of what needs to be reviewed.

H-Shay commented:
I've updated the relative links, PR/issue references, etc, renamed the relevant `.md` files using snake case and added the blurb to the changelog. I will create a follow-up PR to make the suggested changes from  @dklimpel (thank you for the review, btw!).

squahtx commented on CHANGES.md:
```diff
@@ -18,7 +35,7 @@ The following issue is fixed in 1.47.1.
   Homeservers with the media repository disabled are unaffected. Homeservers configured with a federation whitelist are also unaffected.
 
-  Fixed by [91f2bd090](https://github.com/matrix-org/synapse/commit/91f2bd090).
+  Fixed by [91f2bd090](https://github.com/matrix-org/synapse/commit/91f2bd0
```
Was this change intentional?

H-Shay replied:
Not at all, thank you for catching that. I've fixed it.

anoadragon453 commented on CHANGES.md:
```diff
@@ -1,3 +1,19 @@
+in one, easily-searchable location.
+
+Note that contributions to the documentation website happen through [GitHub pull 
+requests](https://github.com/matrix-org/synapse/pulls). Please visit #synapse-dev:matrix.org
```
```suggestion
requests](https://github.com/matrix-org/synapse/pulls). Please visit [#synapse-dev:matrix.org](https://matrix.to/#/#synapse-dev:matrix.org)
```

anoadragon453 commented on CHANGES.md:
```diff
@@ -1,3 +1,19 @@
+==========
+
+We've decided to move the existing, somewhat stagnant pages from the GitHub wiki 
+onto the [documentation website](https://matrix-org.github.io/synapse/latest/).
```
```suggestion
to the [documentation website](https://matrix-org.github.io/synapse/latest/).
```

anoadragon453 reviewed (approved):
Some silly suggestions against my own words, but otherwise this lgtm.

Thanks for creating  #11466 as well!
